### PR TITLE
[OH BOY] Prefs UI Overhaul

### PR DESCRIPTION
**Pull request by RimiNosha**
## About The Pull Request

So, this thing has been like 3 days in the making, and I fucking hate TGUI I hate TGUI I hate TGUI I hate TGUI I hate TGUI.

Anyways, Some prefs (most notably hair gradients, ears, and tails) have been migrated to be icon pickers instead.

Then the main menu for prefs changed to instead be a collection of tabs. This will be extremely important for when we add more prefs later.

Removed the mcolors DNA trait entirely. It's a terrible crutch, and complicates skin color code.

skintone2hex now accepts any colour, along with normal skintones, cause forcing skintones on a race which in this lore genemods the ever living shit out of themselves seems counter-productive.

Prisoner reason selection is visible regardless of prisoner job priority. Not sure why I did this, considering the prisoner job is being removed.

Then I also made species selection require confirmation before fucking with your character, to prevent any broken-ness with mutant parts.

Oh- and it also makes the preference main_feature_name test not immediately return and cancel the test on failing, so you get all your issues at once.

### Known Issues
- Replica pods don't inherit skin colour. Oh wow, oh my god, so big, so bad.
- If you somehow get cleanly species changed via VV, or some other means (remember, magic is gonna be axed), then your skin color might not be appropriate for said species anymore. Not that it was ever guaranteed to be appropriate in the first place.

## Proof of Testing

<details>
<summary>Screenshots/Videos</summary> 

It compiles, give me time to drop a screenshot gallery.
image
image
image
image
image
image
image
image
Misc settings
image

</details>

## Changelog

:cl:
add: Humans now can select ears and tails. Other parts to come later.
qol: Overhauled the prefs menu. There now should be far less information overload for new players in the future.
qol: All species can now pick their exact colors, instead of being restricted to a list. Moths do not use this system, due to them using pre-colored icons.
soundadd: Added a small sound when characters are selected.
soundadd: A misc extra sound for mouseclicks, currently unused.
del: Effectively removes the random and hardcore random prefs.
fix: Cat tails should now be visible when facing south again.
code: De-spaghetti'd skin_tone code.
code: Minimum age is now 18.
del: Removed agender and dyncolor traits.
del: Removed mutant colours and ethereal colours from DNA. Skin_tone now handles that, with species.fixed_mut_color being used for statically colored races (golems).
qol: You can now rotate both directions in preferences.
code: The preference datum is now passed to preference.apply_to_human, allowing them to read other prefs.
fix: Fixes TGUI dev server not working on byond 514.
fix: Pixel scaling is now used for the preview in the prefs UI. Not noticeable at the moment cause we don't allow rescaling yet.
/:cl:

### CODE REVIEW

**Comment by RimiNosha** [comment]
Github enter moment

**Comment by RimiNosha** [comment]
Ready for review. Not adding any new options beyond human ears and tails for now.

**Review by nevimer — APPROVED** [review]
mouse_click.ogg bliss
I'll merge or you can selfmerge or we can discuss and merge tomorrow.

**Comment by nevimer** [comment]
Okay, I toyed around with it. I'd like the ability to do the following in any tab that doesn't hide character preview:

- Change my clothing to naked, underwear, no underwear. 

That's about it, from a quick test. It is kinda inconvenient to deselect your jobs to see your characters body.

**Comment by nevimer** [comment]
dreamdaemon_bZ9rW63HRC

dreamdaemon_AbcvDxYrrn

File: code/modules/client/preferences/species_features/basic.dm [hunk 1]
```diff
@@ -81,20 +81,43 @@
...
 	savefile_identifier = PREFERENCE_CHARACTER
 	savefile_key = "facial_hair_gradient"
 	relevant_species_trait = FACEHAIR
+	should_generate_icons = TRUE
+	main_feature_name = "Facial hair gradient"
 
 /datum/preference/choiced/facial_hair_gradient/init_possible_values()
-	return assoc_to_keys(GLOB.facial_hair_gradients_list)
+	var/list/hair_gradients = list()
+	var/datum/sprite_accessory/hair/hair = GLOB.hairstyles_list["Floorlength Bedhead"]
```
[the remark this answers is not in the document]
Comment (RimiNosha) [reply code/modules/client/preferences/species_features/basic.dm:98]: It isn't, though no facial hair is long enough to show the full gradient, unfortunately.

**Comment by RimiNosha** [comment]
> Change my clothing to naked, underwear, no underwear

Outside the scope of this PR for now.

**Comment by RimiNosha** [comment]
@nevimer Try checking your cat parts issue now.

**Comment by nevimer** [comment]
Compiled and ran, worked good this time.

**Pull request merged**